lib/ex_sdp/attribute.ex

```diff
@@ -12,116 +12,56 @@ defmodule ExSDP.Attribute do
                :keywds,
                :orient,
                :lang,
-               :rtpmap,
                :sdplang,
                :tool,
-               :type
+               :type,
+               :framerate,
+               :maxptime,
+               :ptime,
+               :quality
              ]
 
-  @list_type numeric_attributes :: [:maxptime, :ptime, :quality]
-
-  @type key :: binary() | value_attributes() | numeric_attributes() | :framerate | :rtpmap
+  @type key :: binary() | value_attributes()
   @type value :: binary() | flag_attributes()
-
-  @enforce_keys [:value]
-  @optional_keys [:key]
-
-  defstruct @enforce_keys ++ @optional_keys
-
-  @type t :: %__MODULE__{key: key(), value: value()}
-
-  @flag_values @flag_attributes |> Enum.map(&to_string/1)
-  @directly_assignable_values @value_attributes |> Enum.map(&to_string/1)
-  @numeric_values @numeric_attributes |> Enum.map(&to_string/1)
+  @type t :: __MODULE__.RTPMapping.t() | {key(), value()} | value()
 
   @doc """
-  Parses SDP Attribute.
+  Parses SDP Attribute line.
 
-  `t:value_attributes/0` and `t:numeric_attributes/0` formatted as `name:value`
-  are be parsed as `{name, value}` other values are treated as
-  `t:flag_attributes/0`. Known attribute names are converted into atoms.
-  """
-  @spec parse(binary()) :: {:ok, t} | {:error, atom()}
-  def parse(line) do
-    line
-    |> String.split(":", parts: 2)
-    |> handle_known_attribute()
-  end
+  `line` is a string in form of `a=attribute` or `a=attribute:value`.
+  `opts` is a keyword list that can contain some information for parsers.
 
-  @spec parse_media_attribute(t(), atom() | binary()) :: {:error, :invalid_attribute} | {:ok, t()}
-  def parse_media_attribute(%__MODULE__{key: :rtpmap, value: value}, media) do
-    with {:ok, %RTPMapping{} = mapping} <- RTPMapping.parse(value, media) do
-      {:ok, %__MODULE__{key: :rtpmap, value: mapping}}
-    end
+  Unknown attributes keys are returned as strings, known ones as atoms.
+  """
+  @spec parse(binary(), opts :: []) :: {:ok, t()} | {:error, atom()}
+  def parse(line, opts \\ []) do
+    [attribute | value] = String.split(line, ":", parts: 2)
+    attribute = String.to_atom(attribute)
```

Comment:
converting unknown values to atoms is risky, see https://hexdocs.pm/elixir/String.html#to_atom/1